### PR TITLE
Configure Grouper to load groups from LDAP.

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: src/sharkbait/analyses.clj [hunk 1]
```diff
@@ -1,38 +0,0 @@
-(ns sharkbait.analyses
```
Comment (slr71) [inline src/sharkbait/analyses.clj:1]: This namespace was concerned with permissions in Grouper, which are not used any longer.

File: src/sharkbait/consts.clj [hunk 2]
```diff
@@ -2,17 +2,12 @@
 
 ;; Various folders used by the DE.
 (def folder-format-strings
-  {:de          "iplant:de:%s"
-   :de-users    "iplant:de:%s:users"
-   :de-apps     "iplant:de:%s:apps"
-   :de-analyses "iplant:de:%s:analyses"})
```
Comment (slr71) [inline src/sharkbait/consts.clj:8]: The `apps` and `analyses` folders were intended for permissions.

File: src/sharkbait/core.clj [hunk 3]
```diff
@@ -52,41 +50,18 @@
   (println "Registering DE users...")
   (users/register-de-users session (:de-users folder-names) subjects))
 
-(defn- create-permission-defs
```
Comment (slr71) [inline src/sharkbait/core.clj:55]: More permission stuff.

File: src/sharkbait/db.clj [hunk 4]
```diff
@@ -33,36 +33,6 @@
      :user        user
      :password    (apply str password)}))
 
-(defn- list-de-apps-query
```
Comment (slr71) [inline src/sharkbait/db.clj:36]: Still more permission stuff.

File: src/sharkbait/loader.clj [hunk 5]
```diff
@@ -0,0 +1,38 @@
+(ns sharkbait.loader
```
Comment (slr71) [inline src/sharkbait/loader.clj:1]: This is the namespace that sets up the loader configuration. See https://spaces.internet2.edu/display/Grouper/Grouper+-+Loader+LDAP for more information.

File: test-resources/grouper-loader.base.properties [hunk 6]
```diff
@@ -0,0 +1,314 @@
+# Grouper loader uses Grouper Configuration Overlays (documented on wiki)
```
Comment (slr71) [inline test-resources/grouper-loader.base.properties:1]: These configuration files enable `lein eastwood` to work. They're just copies of the configuration files in `resources` without the references to files in `/etc/grouper`.  Using the configuration files in `resources` causes errors if the files in `/etc/grouper` don't exist.

File: README.md [hunk 7]
```diff
@@ -40,22 +40,30 @@ Grouper's subject source.
...
+* configures Grouper so that it loads groups from LDAP.
 
 Folders:
 
-* iplant:de:analyses
-* iplant:de:apps
-* iplant:de:users
+* `iplant:de:<environment-name>`
+* `iplant:de:<environment-name>:users`
+* `iplant:ldap`
```
Comment (ianmcorvidae) [inline README.md:53]: Probably on your radar but commenting to put it on others' too: we should make sure that LDAP groups are also put into the condor submit files.

File: src/sharkbait/loader.clj [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  (let [delegate (.getAttributeDelegate group)]
+    (.assignAttribute delegate (LoaderLdapUtils/grouperLoaderLdapAttributeDefName))
+    (.retrieveAssignment delegate nil (LoaderLdapUtils/grouperLoaderLdapAttributeDefName) false true)))
+
+(defn- attribute-values
+  "Builds the attribute values for the loader group."
+  [folder-names]
+  {(LoaderLdapUtils/grouperLoaderLdapTypeName)                "LDAP_GROUP_LIST"
+   (LoaderLdapUtils/grouperLoaderLdapFilterName)              "objectClass=posixGroup"
+   (LoaderLdapUtils/grouperLoaderLdapQuartzCronName)          "* * * 1 * * *"
```
Comment (ianmcorvidae) [inline src/sharkbait/loader.clj]: This seems a bit odd. Wouldn't this mean "do this every second of every minute of every hour of the first day of the month"? Guessing something more like `0 0 0 1 * * *`. Similarly, http://www.quartz-scheduler.org/documentation/quartz-1.x/tutorials/crontrigger suggests that the day-of-week field needs to have a `?` in it.

I'm guessing I'm missing something about this string, though. What's going on here?

**Review by ianmcorvidae — APPROVED** [review]
One main question and some other mostly administrative comments. It looks pretty good to me, anyway, so I'm marking Approve.

File: src/sharkbait/loader.clj (hunk 8 above)
Comment (slr71) [reply src/sharkbait/loader.clj]: No, you're right. I'd intended to put zeros in the second, minute and hour fields. Thanks for catching this.

Comment (slr71) [reply src/sharkbait/loader.clj]: Er...the second and minute fields.

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**